textbook/chapter_1.qmd

```diff
@@ -0,0 +1,993 @@
+    x='Miles_per_Gallon',
+    y='Horsepower',
+    tooltip='Name'
+)
+```
+
+
+## Mission
+
+Your name is **Zephyr Eco-Luna** and you have been tasked with investigating emissions in the city of Viridia. With a commitment to sustainability and a vision for a greener future, Viridia is taking proactive steps to investigate and mitigate its environmental impact. One significant area of focus is the emissions associated with food consumption and production.
+
+
+```{=html}
+<br>
+
+        <div class="ministry-id-card">
+            <div class="accent-line"></div>
+            <div class="left-column">
+                <div class="logo">Ministry of Environment</div>
+                <div class="name">Zephyr Eco-Luna</div>
+                <div class="info">
+                    <div class="info-item">
+                        <span class="label">Position</span>
+                        <span class="value">Emissions Investigator</span>
+                    </div>
+                    <div class="info-item">
+                        <span class="label">Department</span>
+                        <span class="value">Urban Food Systems</span>
+                    </div>
+                    <div class="info-item">
+                        <span class="label">ID Number</span>
+                        <span class="value">EI-2024-0042</span>
+                    </div>
+                    <div class="info-item">
+                        <span class="label">Clearance</span>
+                        <span class="value">Level 3</span>
+                    </div>
+                </div>
+                <div class="footer">Protecting Our Future • Est. 2024</div>
+            </div>
+            <div class="right-column">
+                <img src="/img/zephyr.jpg" alt="Avatar" class="avatar">
+            </div>
+        </div>
+<br>
+```
+
+To start with your investigation, you’ll need to visualize the emissions data effectively. This will help unlock insights crucial for the Ministry of Environment.
+
+
+```{python}
+import pandas as pd
+df_food = pd.read_csv("data/aggregated__food_data.csv")
+
+df_food
+```
+
+<img src="/img/environ_minister.jpg" alt="Avatar" class="avatar" style="display: none;">
+
+#### Exercise 1: Planning your analysis
+
+Before you start, your boss **Minister Reed** has a few questions for you...
+
+```{ojs}
+//| echo: false
+{
+    
+const apiUrl = "https://openaiproxy-621feba224c9.herokuapp.com/api/proxy";
+
+const instructions = "You are Minister Reed of the city of Viridia, a sharp-witted, impatient but supportive environmental minister. Your colleague Zephyr has been tasked with creating an Emissions Report. Guide Zephyr toward understanding why a complete picture is necessary and why including the Total Food Supply (kg/capita/yr) column matters (alongside the Emissions Kg CO2 / kg product). Once you get a satisfactory answer, end the conversation immediately. You are in a phone text convo.";
```

Comment:
ok so this genAI is... FANTASTIC! It's so helpful already! I tried a few different answers and it gives a really engaging dialog with helpful hints without giving away everything. If the student fails multiple times over, it seems to explain the full answer in a way that is helpful for the student. So well done already in the first draft @rorywhite200 ! I will look the code in more detail tomorrow but the user experience is already top notch!

Reply:
Yay! Yes I was surprised by how well this works. I think we could create a pretty immersive world with this and it could really complement their learning. Also I think one of the benefits with this approach is that each agent is basically designed for an individual question - they don't need to understand the whole textbook but just be very good at providing hints / dialogue for a single exercise. They are like modular units we can deploy throughout the book and just change the custom instructions. 

I really think this gets us closer to our goal of teaching the fundamentals of how to think about data visualization. Anyway, I'm excited about where this could go. But may be useful to do some back of the envelope calculations about the expected LLM cost per student.